hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -443,12 +443,6 @@ private Dataset<Row> readRecordsForGroupAsRow(JavaSparkContext jsc,
         .toArray(StoragePath[]::new);
 
     HashMap<String, String> params = new HashMap<>();
-    if (hasLogFiles) {
-      params.put("hoodie.datasource.query.type", "snapshot");
-    } else {
-      params.put("hoodie.datasource.query.type", "read_optimized");
-    }
```

Comment:
https://github.com/apache/hudi/blob/master/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala#L250, which shows there is no default value set. 
I did not know if it is safe to rely the default value either.

Meanwhile, do we know how `HadoopFSRelation` ensure no performance degradation for base file-only cases?

Reply:
The config itself has the default value defined but as you pointed out it's not honored through the read path with file paths provided (which the clustering execution uses).
```
val QUERY_TYPE: ConfigProperty[String] = ConfigProperty
    .key("hoodie.datasource.query.type")
    .defaultValue(QUERY_TYPE_SNAPSHOT_OPT_VAL)
    .withAlternatives("hoodie.datasource.view.type")
    .withValidValues(QUERY_TYPE_SNAPSHOT_OPT_VAL, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, QUERY_TYPE_INCREMENTAL_OPT_VAL)
    .withDocumentation("Whether data needs to be read, in `" + QUERY_TYPE_INCREMENTAL_OPT_VAL + "` mode (new data since an instantTime) " +
      "(or) `" + QUERY_TYPE_READ_OPTIMIZED_OPT_VAL + "` mode (obtain latest view, based on base files) (or) `" + QUERY_TYPE_SNAPSHOT_OPT_VAL + "` mode " +
      "(obtain latest view, by merging base and (if any) log files)")
```
Also, after checking the code again, when the file paths are provided through "hoodie.datasource.read.paths", the relation-based read path is used (i.e., `useNewParquetFileFormat` is `false`).  We can keep this for now.

Filed HUDI-8576 as a follow-up.

Reply:
HUDI-8577 to track default change.  We can take this after Hudi 1.0 is released.